controller/instance_manager_controller.go

```diff
@@ -406,10 +406,13 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		if err := imc.cleanupInstanceManager(im); err != nil {
 			return err
 		}
-		if err := imc.createInstanceManagerPod(im); err != nil {
+		created, err := imc.createInstanceManagerPod(im)
```

Comment:
You can use `types.ErrorAlreadyExists(err) ` here for the error instead of passing a boolean.

Since we're using `errors.Wrapf`, you can use `errors.Cause(err)` to unwrap the error.